[<img align="left" width="150" height="150" src="https://github.com/CLIMtools/T-CLIM/blob/main/www/eCLIM.png">](https://rstudio.aws.science.psu.edu:3838/aaf11/T-CLIM/ "T-CLIM")

# [T-CLIM V2.0](https://rstudio.aws.science.psu.edu:3838/aaf11/T-CLIM/ "AraCLIM V2.0")
[**T-CLIM**](https://rstudio.aws.science.psu.edu:3838/aaf11/AraCLIM/) (https://rstudio.aws.science.psu.edu:3838/aaf11/AraCLIM/)is an SHINY component of [**Arabidopsis CLIMtools**](http://www.CLIMtools.org) This tool provides information of the association between the natural variation in the transcript abundance of any gene of interest and either one of the more than 400 environmental variablesn.

TWAS analysis for each of the more than 400 environmental variables in this study was performed using the set 558 accessions within the set of 879 Eurasian accessions in this study that had information on transcript abundance from a previous study (Kawakatsu, Taiji, et al. 2016; DOI: 10.1016/j.cell.2016.06.044). 

Please visit the [Assmann lab](http://www.personal.psu.edu/sma3/) to learn about other interesting research topics.

## [Data availability](https://github.com/CLIMtools/T-CLIM/tree/master/data)

## [Citation](https://www.nature.com/articles/s41559-018-0754-5)
**Ferrero-Serrano, √Å & Assmann SM.** Phenotypic and genome-wide association with the local environment of Arabidopsis. Nature Ecology & Evolution. doi: 10.1038/s41559-018-0754-5 (2019)

[**Read-only link to our paper**](https://www.nature.com/articles/s41559-018-0754-5.epdf?author_access_token=-XAHfjk0t-xpfiAKsAyGWtRgN0jAjWel9jnR3ZoTv0MbnmJteG8gNCxdeNQO1H4w_SmcB99HlQmvk-xORaUBXHbhOHQJv3G6r40i4WmSFedhvQDmjJodK4yv2BRfG7uowtsRL1Za0vHy8o3wpxL4Jw%3D%3D)

[<img align="left" src="https://github.com/CLIMtools/AraCLIM/blob/master/Screen Shot.png">](https://rstudio.aws.science.psu.edu:3838/aaf11/AraCLIM/ "AraCLIM")
